tests/pytest/test_short_read.py

```diff
@@ -0,0 +1,524 @@
+# encoding=utf-8
+import json
+import os
+import random
+import subprocess
+import tempfile
+import zipfile
+
+import gevent.queue
+import gevent.server
+import gevent.socket
+from RLTest import Defaults
+from enum import Enum, auto
+from common import TimeLimit
+
+
+Defaults.decode_responses = True
+
+CREATE_INDICES_TARGET_DIR = '/tmp/test'
+BASE_RDBS_URL = 'https://s3.amazonaws.com/redismodules/redisearch-enterprise/rdbs/'
+
+SHORT_READ_BYTES_DELTA = int(os.getenv('SHORT_READ_BYTES_DELTA', '1'))
+
+RDBS = ['short-reads/rejson_keys_2.0.0.rdb.zip']
+
+
+def unzip(zip_path, to_dir):
+    if not zipfile.is_zipfile(zip_path):
+        return False
+    with zipfile.ZipFile(zip_path, 'r') as db_zip:
+        for info in db_zip.infolist():
+            if not os.path.exists(db_zip.extract(info, to_dir)):
+                return False
+    return True
+
+
+def downloadFiles(target_dir):
+    for f in RDBS:
+        path = os.path.join(target_dir, f)
+        path_dir = os.path.dirname(path)
+        if not os.path.exists(path_dir):
+            os.makedirs(path_dir)
+        if not os.path.exists(path):
+            subprocess.call(['wget', '-q', BASE_RDBS_URL + f, '-O', path])
+            _, ext = os.path.splitext(f)
+            if ext == '.zip':
+                if not unzip(path, path_dir):
+                    return False
+            else:
+                if not os.path.exists(path) or os.path.getsize(path) == 0:
+                    return False
+    return True
+
+
+def rand_name(k):
+    # rand alphabetic string with between 2 to k chars
+    return ''.join([chr(random.randint(ord('a'), ord('z'))) for _ in range(0, random.randint(2, max(2, k)))])
```

Comment:
We should use some random generate seed and log it so if the test failed it will be easier to reproduce it.